src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (!Animal.alive) {
+      Animal.alive = [];
+    }
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  bite(target) {
+    if (target instanceof Carnivore
+      || (target instanceof Herbivore && target.hidden)) {
+      return;
+    }
+
+    target.health -= 50;
```

Comment:
```suggestion
    if (target instanceof Herbivore && !target.hidden) {
          target.health -= 50;
    }
```

we can simplify it a bit